ee/tabby-db/src/github_repository_provider.rs

```diff
@@ -0,0 +1,50 @@
+use anyhow::{anyhow, Result};
+use sqlx::{query, query_as};
+
+use crate::{DbConn, SQLXResultExt};
+
+pub struct GithubProviderDAO {
+    pub display_name: String,
+    pub application_id: String,
+    pub secret: String,
+}
+
+impl DbConn {
+    pub async fn update_github_provider(
+        &self,
+        name: String,
+        application_id: String,
+        secret: String,
+    ) -> Result<()> {
+        query!("INSERT INTO github_repository_provider (display_name, application_id, secret) VALUES ($1, $2, $3);",
+            name,
+            application_id,
+            secret
+        ).execute(&self.pool).await.unique_error("Provider already exists")?;
```

Comment:
is this will be something might displayed in frontend, it should really be something like "Application ID already exists"